src/nvim/terminal.c

```diff
@@ -465,12 +453,8 @@ static int terminal_execute(VimState *state, int key)
       break;
 
     case K_EVENT:
-      // We cannot let an event free the terminal yet. It is still needed.
-      s->term->refcount++;
       multiqueue_process_events(main_loop.events);
-      s->term->refcount--;
```

Comment:
The only real purpose this ref counter had was to distinguish if `terminal_close()` was called due to a job that exited or the buffer being wiped. Since we now have separate functions for those (`terminal_close()` and `terminal_exit()`), it should be no longer required.